core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java

```diff
@@ -414,32 +419,61 @@ private void decoratePillarWithClouderaManagerAutoTls(Cluster cluster, Map<Strin
         }
     }
 
-    private void decoratePillarWithClouderaManagerLicense(Long stackId, Map<String, SaltPillarProperties> servicePillar) {
+    private Optional<String> decoratePillarWithClouderaManagerLicense(Long stackId, Map<String, SaltPillarProperties> servicePillar) {
         String userCrn = stackService.get(stackId).getCreator().getUserCrn();
-
         Account account = umsClient.getAccountDetails(userCrn, Crn.safeFromString(userCrn).getAccountId(), Optional.empty());
-
-        if (StringUtils.isNotEmpty(account.getClouderaManagerLicenseKey())) {
-            LOGGER.debug("Got license key from UMS: {}", account.getClouderaManagerLicenseKey());
+        Optional<String> licenseOpt = Optional.ofNullable(account.getClouderaManagerLicenseKey());
+        if (licenseOpt.isPresent() && isNotEmpty(licenseOpt.get())) {
+            String license = licenseOpt.get();
+            LOGGER.debug("Got license key from UMS: {}", license);
             servicePillar.put("cloudera-manager-license",
                     new SaltPillarProperties("/cloudera-manager/license.sls",
                             singletonMap("cloudera-manager",
-                                    singletonMap("license", account.getClouderaManagerLicenseKey()))));
+                                    singletonMap("license", license))));
         }
+        return licenseOpt;
     }
 
-    private void decoratePillarWithClouderaManagerRepo(Long stackId, Long clusterId, Map<String, SaltPillarProperties> servicePillar)
+    @VisibleForTesting
+    void decoratePillarWithClouderaManagerRepo(Long clusterId, Map<String, SaltPillarProperties> servicePillar, Optional<String> license)
             throws CloudbreakOrchestratorFailedException {
-
         ClouderaManagerRepo clouderaManagerRepo = clusterComponentConfigProvider.getClouderaManagerRepoDetails(clusterId);
-
         if (clouderaManagerRepo == null) {
             throw new CloudbreakOrchestratorFailedException("Cloudera Manager repository details are missing.");
         }
-
         servicePillar.put("cloudera-manager-repo", new SaltPillarProperties("/cloudera-manager/repo.sls",
-                singletonMap("cloudera-manager", singletonMap("repo", clouderaManagerRepo))));
+                singletonMap("cloudera-manager", createCMRepoPillar(clouderaManagerRepo, license))));
+
+    }
 
+    private Map<String, Object> createCMRepoPillar(ClouderaManagerRepo clouderaManagerRepo, Optional<String> license) {
+        Map<String, Object> pillarValues = new HashMap<>();
+        pillarValues.put("repo", clouderaManagerRepo);
+        parseLicense(license).ifPresent(jsonLicense -> {
+            String username = jsonLicense.getPaywallUsername();
+            String password = jsonLicense.getPaywallPassword();
+            if (isNotEmpty(username) && isNotEmpty(password)) {
```

Comment:
just mentioning, don't have to replace: `StringUtils#isNoneEmpty` could work also